frontend/components/ManageStakingPage/StakingContractSection/CantMigrateAlert.tsx

```diff
@@ -20,33 +27,65 @@ const { Text } = Typography;
 type CantMigrateAlertProps = { stakingProgramId: StakingProgramId };
 
 const AlertInsufficientMigrationFunds = ({
-  stakingProgramId,
+  stakingProgramId: stakingProgramIdToMigrateTo,
 }: CantMigrateAlertProps) => {
-  const { serviceTemplate } = useServiceTemplates();
+  const {
+    isFetched: isServicesLoaded,
+    selectedService,
+    selectedAgentConfig,
+  } = useServices();
+  const { homeChainId } = selectedAgentConfig;
+  const serviceConfigId =
+    isServicesLoaded && selectedService
+      ? selectedService.service_config_id
+      : '';
+  const { service } = useService({
+    serviceConfigId,
+  });
   const { isAllStakingContractDetailsRecordLoaded } =
     useStakingContractContext();
   const { isServiceStaked } = useActiveStakingContractInfo();
-  const { masterSafeBalance: safeBalance, totalOlasStakedBalance } =
+  const { isLoaded: isBalanceLoaded, totalStakedOlasBalance } =
     useBalanceContext();
+  const { masterSafeBalances } = useMasterBalances();
   const { serviceFundRequirements, isInitialFunded } = useNeedsFunds();
 
-  const totalOlasRequiredForStaking = getMinimumStakedAmountRequired(
-    serviceTemplate,
-    stakingProgramId,
-  );
+  // should find in STAKING_PROGRAMS based on stakingProgramIdToMigrateTo?
+  const chainIdToMigrateTo = ChainId.Gnosis;
+
+  const requiredStakedOlas =
+    service &&
```

Comment:
copy-pasted this from AgentNotRunningButton but wonder should we check service

Reply:
can check this without service condition